usaspending_api/bulk_download/management/commands/populate_monthly_files.py

```diff
@@ -0,0 +1,211 @@
+import logging
+import datetime
+import json
+import multiprocessing
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from usaspending_api.bulk_download.filestreaming import csv_selection
+from usaspending_api.bulk_download.v2.views import BulkDownloadAwardsViewSet
+from usaspending_api.common.helpers import generate_fiscal_year
+from usaspending_api.common.csv_helpers import sqs_queue
+from usaspending_api.bulk_download.models import BulkDownloadJob
+from usaspending_api.download.lookups import JOB_STATUS_DICT
+from usaspending_api.references.models import ToptierAgency, SubtierAgency
+from usaspending_api.common.helpers import order_nested_object
+
+# Logging
+logger = logging.getLogger('console')
+
+award_mappings = {
+    'contracts': ['contracts'],
+    'assistance': ['grants', 'direct_payments', 'loans', 'other_financial_assistance']
+}
+
+
+class Command(BaseCommand):
+    # TODO: This function reveals most of bulk download relies on its REST origins.
+    #       This and most of Bulk Download will definitely need to be refactored to address this.
+    def bulk_download(self, file_name, award_levels, award_types=None, agency=None, sub_agency=None,
+                      date_type=None, start_date=None, end_date=None, columns=[], file_format="csv",
+                      monthly_download=False, use_sqs=False):
+        date_range = {}
+        if start_date:
+            date_range['start_date'] = start_date
+        if end_date:
+            date_range['end_date'] = end_date
+        json_request = {'award_levels': award_levels,
+                        'filters': {
+                            'award_types': award_types,
+                            'agency': agency,
+                            'date_type': date_type,
+                            'date_range': date_range,
+                        },
+                        'columns': columns,
+                        'file_format': file_format}
+        sources = BulkDownloadAwardsViewSet().get_csv_sources(json_request=json_request)
+
+        download_job_kwargs = {'job_status_id': JOB_STATUS_DICT['ready'],
+                               'monthly_download': monthly_download,
+                               'json_request': json.dumps(order_nested_object(json_request)),
+                               'file_name': file_name,
+                               'date_type': date_type}
+        for award_level in award_levels:
+            download_job_kwargs[award_level] = True
+        for award_type in award_types:
+            download_job_kwargs[award_type] = True
+        if agency and agency != 'all':
+            download_job_kwargs['agency'] = ToptierAgency.objects.filter(toptier_agency_id=agency).first()
+        if sub_agency:
+            download_job_kwargs['sub_agency'] = SubtierAgency.objects.filter(subtier_agency_id=sub_agency).first()
+        if start_date:
+            download_job_kwargs['start_date'] = start_date
```

Comment:
Does this default to all FYs?

Reply:
In this function, if start_date and end_date are not provided, it assumed it will cover all FYs. The way this function is called currently is that the start/end dates are set to each fiscal year (see lines 194-211)

Reply:
makes sense